pkg/operator/imageconfig.go

```diff
@@ -0,0 +1,251 @@
+// the resource status appropriately.
+type ImageConfigController struct {
+	configClient  configset.ConfigV1Interface
+	routeLister   routev1lister.RouteNamespaceLister
+	serviceLister corev1listers.ServiceNamespaceLister
+	cachesToSync  []cache.InformerSynced
+	queue         workqueue.RateLimitingInterface
+}
+
+func NewImageConfigController(
+	configClient configset.ConfigV1Interface,
+	routeInformer routev1informers.RouteInformer,
+	serviceInformer corev1informers.ServiceInformer,
+) *ImageConfigController {
+	icc := &ImageConfigController{
+		configClient: configClient,
+		routeLister: routeInformer.Lister().Routes(
+			defaults.ImageRegistryOperatorNamespace,
+		),
+		serviceLister: serviceInformer.Lister().Services(
+			defaults.ImageRegistryOperatorNamespace,
+		),
+		queue: workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(),
+			"ImageConfigController",
+		),
+	}
+
+	serviceInformer.Informer().AddEventHandler(icc.eventHandler())
+	icc.cachesToSync = append(
+		icc.cachesToSync, serviceInformer.Informer().HasSynced,
+	)
+
+	routeInformer.Informer().AddEventHandler(icc.eventHandler())
+	icc.cachesToSync = append(
+		icc.cachesToSync, routeInformer.Informer().HasSynced,
+	)
+
+	return icc
+}
+
+func (icc *ImageConfigController) eventHandler() cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			icc.queue.Add("instance")
+		},
+		UpdateFunc: func(old, new interface{}) {
+			icc.queue.Add("instance")
+		},
+		DeleteFunc: func(obj interface{}) {
+			icc.queue.Add("instance")
+		},
+	}
+}
+
+func (icc *ImageConfigController) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer icc.queue.ShutDown()
+
+	klog.Infof("Starting ImageConfigController")
+	if !cache.WaitForCacheSync(stopCh, icc.cachesToSync...) {
+		return
+	}
+
+	go wait.Until(icc.runWorker, time.Second, stopCh)
+
+	klog.Infof("Started ImageConfigController")
+	<-stopCh
+	klog.Infof("Shutting down ImageConfigController")
+}
+
+func (icc *ImageConfigController) runWorker() {
+	for icc.processNextWorkItem() {
+	}
+}
+
+func (icc *ImageConfigController) processNextWorkItem() bool {
+	obj, shutdown := icc.queue.Get()
+	if shutdown {
+		return false
+	}
+	defer icc.queue.Done(obj)
+
+	klog.V(1).Infof("get event from workqueue")
+	if err := icc.sync(); err != nil {
+		icc.queue.AddRateLimited(workqueueKey)
+		klog.Errorf("ImageConfigController: unable to sync: %s, requeuing", err)
+	} else {
+		icc.queue.Forget(obj)
+		klog.Infof("ImageConfigController: event from workqueue processed")
+	}
+
+	return true
+}
+
+// sync keeps image.config.openshift.io/cluster status updated.
+func (icc *ImageConfigController) sync() error {
+	cfg, err := icc.configClient.Images().Get(
+		context.TODO(), defaults.ImageConfigName, metav1.GetOptions{},
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	// This resource must always exist.
```

Comment:
What does this comment mean?

Reply:
Mental note I forgot to remove. Removed now.